benchmark-overhead-jmh/src/main/java/io/opentelemetry/javaagent/benchmark/util/HttpClient.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.benchmark.util;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.List;
+
+// this minimal http client really helps to keep benchmark run variation to a minimum
+public class HttpClient {
+
+  private static final int LOWERCASE_OFFSET = 'a' - 'A';
+
+  private static final byte[] CONTENT_LENGTH_BYTES = "CONTENT-LENGTH:".getBytes(UTF_8);
+
+  private final Socket socket;
+  private final OutputStream out;
+  private final InputStream in;
+  private final byte[] requestBytes;
+
+  private final byte[] buffer = new byte[8192];
+
+  public HttpClient(String host, int port, String path) throws IOException {
+    socket = new Socket(host, port);
```

Comment:
I'm assuming you've already found this Socket approach to have less variance than HttpUrlConnection?

Reply:
oh, I had not!

I was initially trying to use an http client that wouldn't get instrumented, but that's easy to workaround with `otel.instrumentation.http-url-connection.enabled=false`.

Here's results with `HttpURLConnection` (which are inline variance-wise with ugly Socket approach 💯):

```
Iteration   1: 106.494 ±(99.9%) 0.320 us/op
Iteration   2: 107.312 ±(99.9%) 0.324 us/op
Iteration   3: 105.766 ±(99.9%) 0.314 us/op
Iteration   4: 105.894 ±(99.9%) 0.320 us/op
Iteration   5: 107.098 ±(99.9%) 0.335 us/op
Iteration   6: 105.891 ±(99.9%) 0.315 us/op
Iteration   7: 104.920 ±(99.9%) 0.310 us/op
Iteration   8: 106.473 ±(99.9%) 0.307 us/op
Iteration   9: 106.503 ±(99.9%) 0.323 us/op
Iteration  10: 106.765 ±(99.9%) 0.315 us/op
Iteration  11: 106.019 ±(99.9%) 0.335 us/op
Iteration  12: 104.807 ±(99.9%) 0.305 us/op
Iteration  13: 107.513 ±(99.9%) 0.309 us/op
Iteration  14: 106.557 ±(99.9%) 0.322 us/op
Iteration  15: 106.545 ±(99.9%) 0.321 us/op
Iteration  16: 106.283 ±(99.9%) 0.315 us/op
Iteration  17: 106.466 ±(99.9%) 0.340 us/op
Iteration  18: 106.274 ±(99.9%) 0.324 us/op
Iteration  19: 105.127 ±(99.9%) 0.309 us/op
Iteration  20: 106.024 ±(99.9%) 0.311 us/op
```